match/lib/match/git_helper.rb

```diff
@@ -43,6 +45,29 @@ def self.clone(git_url,
         UI.user_error!("Error cloning certificates git repo, please make sure you have access to the repository - see instructions above")
       end
 
+      # if there is a encryption state marker file
+      if File.exist?(File.join(@dir, "match_crypted.txt"))
+        # repo is crypted but user requested to run match in disabled_encryption mode
+        # files cannot be read.
+        if GitHelper.encrypted?(@dir) && disable_encryption
+          remote_is_encrypted!
+        end
+
+        # repo is not crypted, but match is run with enabled encryption
+        if !GitHelper.encrypted?(@dir) && !disable_encryption
+          UI.error("Encryption enabled, but remote repository is currently decrypted.")
+          UI.error("See: https://github.com/fastlane/fastlane/pull/8919 for details on how to convert your existing repository")
+          UI.user_error!("remote_decrypted")
+        end
+      else
+        # existing repo has `match_version.txt` - but no encryption state marker but unencrypted request
+        if File.exist?(File.join(@dir, "match_version.txt")) && disable_encryption
+          UI.error("You requested to disable encryption on a repo that is not up-to-date")
+          UI.error("Please run match atleast once with the current version")
+          remote_is_encrypted!
+        end
+      end
```

Comment:
This method is pretty long with a lot of special cases, could we extract some stuff out into separate, methods that could be tested? That might also allow us to remove the rubocop perceived complexity warning suppression as well.